cachi2/core/package_managers/pip.py

```diff
@@ -136,6 +139,11 @@ def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, str]:
         name = name or setup_cfg.get_name()
         version = version or setup_cfg.get_version()
 
+    if not (name and version) and pyproject_toml.exists():
+        log.info("Filling in missing metadata from pyproject.toml")
+        name = pyproject_toml.get_name()
+        version = pyproject_toml.get_version()
```

Comment:
I would make `pyproject.toml` superior in this case, since `setup.py` is deprecated. Also I played around with it and if both files are present in the project folder, pip takes `pyproject.toml` first to build package.